### DISCUSSION TITLE
What should I do to make getting objects from video (YouTube) faster?

### DISCUSSION THREAD

**Discussion by KatarinaPopikova** [ultralytics/ultralytics#1425] [Q&A]
Hello!
All I need is to get the objects that appeared in the video (YouTube).
So I tried set attribute vid_stride to 50 and imgsz=320, but it takes almost the same time. I also tried device="cpu"  and I got the same result in computing time. 

I tried this:

`model.predict(source=SOURCE, device="cpu", verbose=False)`

`model.predict(source=SOURCE, device=0, vid_stride=50, verbose=False, imgsz=320)`

What should I do to make getting objects from **video faster**?

**Comment by glenn-jocher** [comment]
@KatarinaPopikova hello! To speed up object detection in videos from YouTube using YOLOv8, consider the following suggestions:

1. **Reduce Frame Rate**: Increasing `vid_stride` skips more frames, which should reduce processing time. Try setting it even higher if you can afford to miss more frames.

2. **Lower Resolution**: You've already tried reducing `imgsz` to 320. You might experiment with going even lower, but be aware that this can affect the accuracy of detection.

3. **Batch Processing**: If YOLOv8 supports batch processing for videos, process multiple frames at once to utilize your GPU more efficiently.

4. **Hardware Acceleration**: Ensure you're using a GPU if available (`device=0` typically refers to the first GPU). The CPU (`device="cpu"`) is generally slower.

5. **Optimized Model**: Use a smaller and faster model variant if available, which can provide a good balance between speed and accuracy.

6. **Parallel Processing**: If you're processing multiple videos, you can run them in parallel using multiprocessing techniques to utilize all CPU/GPU resources.

7. **Profile Your Model**: Use the Benchmark mode to profile your model and identify bottlenecks.

Remember that each video and use case may require a different combination of these optimizations for the best results. 🚀
